manager/volume.go

```diff
@@ -614,7 +617,59 @@ func (m *VolumeManager) Expand(volumeName string, size int64) (v *longhorn.Volum
 	}
 
 	if v.Spec.Size > size {
-		return nil, fmt.Errorf("cannot expand volume %v with current size %v to a smaller size %v", v.Name, v.Spec.Size, size)
+		return nil, fmt.Errorf("longhorn volume %v spec size %v is greater than requested Persistent Volume Claim size %v", v.Name, v.Spec.Size, size)
+
+	}
+
+	if v.Status.ActualSize > size {
+		return nil, fmt.Errorf("longhorn volume %v actual size %v is greater than requested Persistent Volume Claim size %v", v.Name, v.Spec.Size, size)
+	}
+
+	kubernetesStatus := &v.Status.KubernetesStatus
+	if kubernetesStatus.PVCName != "" && kubernetesStatus.LastPVCRefAt == "" {
+		pvc, err := m.ds.GetPersistentVolumeClaim(kubernetesStatus.Namespace, kubernetesStatus.PVCName)
+		if err != nil {
+			return nil, err
+		}
+
+		requestedSize := resource.MustParse(strconv.FormatInt(size, 10))
+
+		// TODO: Should check for pvc.Spec.Resources.Requests.Storage() here, once upgrade API to v0.18.x.
+		storageSpecValue, ok := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
+		if !ok {
+			return nil, fmt.Errorf("cannot get request storage")
+		}
+		// TODO: Should check for pvc.Status.CapacityStorage() here, once upgrade API to v0.18.x.
+		storageStatusValue, ok := pvc.Status.Capacity[corev1.ResourceStorage]
+		if !ok {
+			return nil, fmt.Errorf("cannot get capacity storage")
+		}
+
+		if storageSpecValue.Cmp(requestedSize) == 0 && storageStatusValue.Cmp(requestedSize) == 0 && v.Spec.Size == size {
+			logrus.Infof("Persistent Volume Claim %v expansion is not necessary since current size %v == %v", v.Name, v.Spec.Size, size)
+			return v, nil
+		} else if storageSpecValue.Cmp(requestedSize) > 0 {
+			return nil, fmt.Errorf("persistent volume claim %v spec size %v is greater than requested persistent volume claim size %v", kubernetesStatus.PVCName, storageSpecValue.Value(), requestedSize.Value())
+		} else if storageStatusValue.Cmp(requestedSize) > 0 {
+			return nil, fmt.Errorf("persistent volume claim %v status size %v is greater than requested persistent volume claim size %v", kubernetesStatus.PVCName, storageStatusValue.Value(), requestedSize.Value())
+		} else if storageSpecValue.Cmp(requestedSize) < 0 {
```

Comment:
Just need to `else {` good enough. 

nit: for multiple if-else, you can probably use `switch` w/ no condition statement.

Reply:
removed else